src/main/kotlin/no/nav/helsearbeidsgiver/kafka/forespoersel/ForespoerselTolker.kt

```diff
@@ -4,79 +4,80 @@ import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import no.nav.helsearbeidsgiver.forespoersel.ForespoerselDokument
 import no.nav.helsearbeidsgiver.forespoersel.ForespoerselRepository
-import no.nav.helsearbeidsgiver.kafka.LpsKafkaConsumer
+import no.nav.helsearbeidsgiver.kafka.MeldingTolker
 import no.nav.helsearbeidsgiver.mottak.ExposedMottak
 import no.nav.helsearbeidsgiver.mottak.MottakRepository
 import no.nav.helsearbeidsgiver.utils.jsonMapper
+import org.jetbrains.exposed.sql.transactions.transaction
 import org.slf4j.LoggerFactory
 
-class ForespoerselKafkaConsumer(
+class ForespoerselTolker(
     private val forespoerselRepository: ForespoerselRepository,
     private val mottakRepository: MottakRepository,
-) : LpsKafkaConsumer {
+) : MeldingTolker {
     private val sikkerLogger = LoggerFactory.getLogger("tjenestekall")
 
-    override fun handleRecord(record: String) {
-        // transaction {
-        sikkerLogger.info("Mottatt event: $record")
-        val obj = parseRecord(record)
-        if (obj == null) {
-            sikkerLogger.warn("Ugyldig event mottatt: $record")
-            mottakRepository.opprett(ExposedMottak(inntektsMelding = record, gyldig = false))
-            return
-        }
+    override fun lesMelding(melding: String) {
+        sikkerLogger.info("Mottatt event: $melding")
+        val obj =
+            try {
+                parseRecord(melding)
+            } catch (e: Exception) {
+                sikkerLogger.info("Ugyldig event, ignorerer melding")
+                mottakRepository.opprett(ExposedMottak(melding = melding, gyldig = false))
+                return
+            }
         try {
-            sikkerLogger.info("Received notis: ${obj.notis}")
+            sikkerLogger.info("Mottatt notis: ${obj.notis}")
 
             when (obj.notis) {
                 NotisType.FORESPØRSEL_MOTTATT -> {
                     val forespoersel = obj.forespoersel
                     if (forespoersel != null) {
-                        forespoerselRepository.lagreForespoersel(
-                            forespoerselId = forespoersel.forespoerselId.toString(),
-                            payload = forespoersel,
-                        )
-                        mottakRepository.opprett(ExposedMottak(record))
+                        transaction {
+                            try {
+                                forespoerselRepository.lagreForespoersel(
+                                    forespoerselId = forespoersel.forespoerselId.toString(),
+                                    payload = forespoersel,
+                                )
+                                mottakRepository.opprett(ExposedMottak(melding))
+                            } catch (e: Exception) {
+                                rollback()
```

Comment:
Litt usikker her. Vil koden etter rollback kjøre? Vil ikke transaksjonen rolles tilbake når du kaster exception uansett?

Reply:
koden vil kjøre ja. Dette er for å sørge for at enten begge eller ingen inserts gjøres i database, så vi ikke får delvis commits.